tests/unit/UrlSearchParams.ts

```diff
@@ -0,0 +1,130 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import UrlSearchParams from 'src/UrlSearchParams';
+
+registerSuite({
+	name: 'UrlSearchParams',
+
+	construct: (function () {
+		function checkParams(params: UrlSearchParams) {
+			assert.deepEqual(params.getAll('foo'), [ 'bar', 'baz' ]);
+			assert.deepEqual(params.getAll('bar'), [ 'foo' ]);
+			assert.deepEqual(params.getAll('baz'), [ '' ]);
+		}
+
+		return {
+			empty() {
+				// Empty UrlSearchParams should have no properties
+				const params = new UrlSearchParams();
+				assert.isNull(params.get('foo'));
+			},
+
+			// Next three tests should generate 'params' objects with same contents
+
+			object() {
+				const params = new UrlSearchParams({ foo: [ 'bar', 'baz' ], bar: 'foo', baz: null });
+				checkParams(params);
+			},
+
+			string() {
+				let params = new UrlSearchParams('foo=bar&foo=baz&bar=foo&baz');
+				checkParams(params);
+
+				// Assignments after the first will be ignored
+				params = new UrlSearchParams('foo=bar=baz');
+				assert.deepEqual(params.getAll('foo'), [ 'bar' ]);
+
+				// Handle empty keys
+				params = new UrlSearchParams('=foo&');
+				assert.deepEqual(params.getAll(''), [ 'foo', '' ]);
+			},
+
+			UrlSearchParams() {
```

Comment:
Should we also have a test somewhere to verify that the two instances do not share values?